addons/misra.py

```diff
@@ -840,13 +841,19 @@ def isFloatCounterInWhileLoop(whileToken):
     token = whileBodyStart
     while token != whileBodyStart.link:
         token = token.next
-        for counterToken in counterTokens:
-            if not counterToken.valueType or not counterToken.valueType.isFloat():
-                continue
-            if token.isAssignmentOp and token.astOperand1.str == counterToken.str:
+        if token.str not in tok_str:
+            continue
+        if not token.astParent or not token.valueType or not token.valueType.isFloat():
+            continue
+        parent = token.astParent
+        if parent.str in ('++', '--'):
+            return True
+        while parent:
+            if parent.isAssignmentOp and parent.str != "=":
```

Comment:
this does not feel proper to me. Imagine that token.str is `x`. and the expression is `y += x;`. I don't think that we should return true here then. It would be good to track if token.str is in the LHS or RHS of the assignment.